mpOpenAPI-4.0 fails to parse a static openapi.json file containing tabs

Found during SVT:

Test with build  [openliberty-25.0.0.1-beta-cl241220241107-1102.zip](https://rtpgsa.ibm.com/projects/l/liberty.build/liberty/WASX.LIBERTY/daily/cl241220241107-1102.30.linux/linux/zipper/externals/installables/openliberty-25.0.0.1-beta-cl241220241107-1102.zip)

I see the following error in the log

> [ERROR   ] CWWKO1661E: An error occurred when processing application SocialSphere-mp and an OpenAPI document was not produced. The error was: io.smallrye.openapi.runtime.OpenApiRuntimeException: Failed to read YAML stream.

_Originally posted by @tam512 in https://github.com/OpenLiberty/open-liberty/issues/29035#issuecomment-2463714620_

When looking in the FFDC files, the full error is revealed to be:

```
io.smallrye.openapi.runtime.OpenApiRuntimeException: Failed to read YAML stream
	at io.smallrye.openapi.runtime.io.JacksonJsonIO.fromReader(JacksonJsonIO.java:385)
	at io.smallrye.openapi.runtime.io.JacksonJsonIO.fromReader(JacksonJsonIO.java:33)
	at io.smallrye.openapi.runtime.io.JsonIO.fromReader(JsonIO.java:473)
	at io.smallrye.openapi.api.SmallRyeOpenAPI$Builder.addStaticModel(SmallRyeOpenAPI.java:588)
	at io.smallrye.openapi.api.SmallRyeOpenAPI$Builder.lambda$buildStaticModel$5(SmallRyeOpenAPI.java:570)
	at io.smallrye.common.classloader.ClassPathUtils.lambda$consumeStream$1(ClassPathUtils.java:170)
	at io.smallrye.common.classloader.ClassPathUtils.readStream(ClassPathUtils.java:197)
	at io.smallrye.common.classloader.ClassPathUtils.consumeStream(ClassPathUtils.java:169)
	at io.smallrye.openapi.api.SmallRyeOpenAPI$Builder.buildStaticModel(SmallRyeOpenAPI.java:570)
	at io.smallrye.openapi.api.SmallRyeOpenAPI$Builder.build(SmallRyeOpenAPI.java:700)
	at io.openliberty.microprofile.openapi40.internal.services.impl.ModelGenerator40Impl.generateModel(ModelGenerator40Impl.java:47)
	at io.openliberty.microprofile.openapi20.internal.ApplicationProcessor.generateModel(ApplicationProcessor.java:323)
	at io.openliberty.microprofile.openapi20.internal.ApplicationProcessor.processWebModule(ApplicationProcessor.java:263)
	at io.openliberty.microprofile.openapi20.internal.ApplicationProcessor.processApplication(ApplicationProcessor.java:145)
	at io.openliberty.microprofile.openapi20.internal.ApplicationRegistryImpl.addApplication(ApplicationRegistryImpl.java:113)
	at io.openliberty.microprofile.openapi20.internal.ApplicationListener.applicationStarting(ApplicationListener.java:48)
	at com.ibm.ws.container.service.state.internal.ApplicationStateManager.fireStarting(ApplicationStateManager.java:53)
	at com.ibm.ws.container.service.state.internal.StateChangeServiceImpl.fireApplicationStarting(StateChangeServiceImpl.java:52)
	at com.ibm.ws.app.manager.module.internal.SimpleDeployedAppInfoBase.preDeployApp(SimpleDeployedAppInfoBase.java:623)
	at com.ibm.ws.app.manager.module.internal.SimpleDeployedAppInfoBase.installApp(SimpleDeployedAppInfoBase.java:584)
	at com.ibm.ws.app.manager.module.internal.DeployedAppInfoBase.deployApp(DeployedAppInfoBase.java:625)
	at com.ibm.ws.app.manager.war.internal.WARApplicationHandlerImpl.install(WARApplicationHandlerImpl.java:67)
	at com.ibm.ws.app.manager.internal.statemachine.StartAction.execute(StartAction.java:199)
	at com.ibm.ws.app.manager.internal.statemachine.ApplicationStateMachineImpl.enterState(ApplicationStateMachineImpl.java:1369)
	at com.ibm.ws.app.manager.internal.statemachine.ApplicationStateMachineImpl.run(ApplicationStateMachineImpl.java:912)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:298)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:839)
Caused by: com.fasterxml.jackson.dataformat.yaml.snakeyaml.error.MarkedYAMLException: while scanning for the next token
found character '\t(TAB)' that cannot start any token. (Do not use \t(TAB) for indentation)
 in 'reader', line 2, column 1:
    	"openapi": "3.0.0",
    ^

 at [Source: (InputStreamReader); line: 1, column: 1]
	at com.fasterxml.jackson.dataformat.yaml.snakeyaml.error.MarkedYAMLException.from(MarkedYAMLException.java:28)
	at com.fasterxml.jackson.dataformat.yaml.YAMLParser.nextToken(YAMLParser.java:433)
	at com.fasterxml.jackson.databind.ObjectMapper._readTreeAndClose(ObjectMapper.java:4854)
	at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:3208)
	at io.smallrye.openapi.runtime.io.JacksonJsonIO.fromReader(JacksonJsonIO.java:382)
	... 28 more
Caused by: while scanning for the next token
found character '\t(TAB)' that cannot start any token. (Do not use \t(TAB) for indentation)
 in 'reader', line 2, column 1:
    	"openapi": "3.0.0",
    ^

	at org.yaml.snakeyaml.scanner.ScannerImpl.fetchMoreTokens(ScannerImpl.java:445)
	at org.yaml.snakeyaml.scanner.ScannerImpl.checkToken(ScannerImpl.java:238)
	at org.yaml.snakeyaml.parser.ParserImpl$ParseImplicitDocumentStart.produce(ParserImpl.java:212)
	at org.yaml.snakeyaml.parser.ParserImpl.peekEvent(ParserImpl.java:162)
	at org.yaml.snakeyaml.parser.ParserImpl.getEvent(ParserImpl.java:172)
	at com.fasterxml.jackson.dataformat.yaml.YAMLParser.nextToken(YAMLParser.java:429)
	... 31 more
```

The file being parsed in this case starts:

```json
{
	"openapi": "3.0.0",
```

The file uses a mix of spaces and tabs for indentation.

This should parse correctly as JSON because the whitespace in this location is not meaningful.

--- a/dev/cnf/dependabot/check_this_in_if_it_changes/pom.xml
+++ b/dev/cnf/dependabot/check_this_in_if_it_changes/pom.xml
@@ -1239,7 +1239,7 @@
     <dependency>
       <groupId>io.smallrye</groupId>
       <artifactId>smallrye-open-api-core</artifactId>
-      <version>4.0.1</version>
+      <version>4.0.2</version>
     </dependency>
     <dependency>
       <groupId>io.smallrye</groupId>
@@ -1249,12 +1249,12 @@
     <dependency>
       <groupId>io.smallrye</groupId>
       <artifactId>smallrye-open-api-jaxrs</artifactId>
-      <version>4.0.1</version>
+      <version>4.0.2</version>
     </dependency>
     <dependency>
       <groupId>io.smallrye</groupId>
       <artifactId>smallrye-open-api-model</artifactId>
-      <version>4.0.1</version>
+      <version>4.0.2</version>
     </dependency>
     <dependency>
       <groupId>io.zipkin.reporter2</groupId>

--- a/dev/cnf/oss_dependencies.maven
+++ b/dev/cnf/oss_dependencies.maven
@@ -243,10 +243,10 @@ io.smallrye:smallrye-graphql-servlet:1.0.26
 io.smallrye:smallrye-graphql:1.0.26
 io.smallrye:smallrye-metrics:5.1.0
 io.smallrye:smallrye-open-api-core:3.1.1
-io.smallrye:smallrye-open-api-core:4.0.1
+io.smallrye:smallrye-open-api-core:4.0.2
 io.smallrye:smallrye-open-api-jaxrs:3.1.1
-io.smallrye:smallrye-open-api-jaxrs:4.0.1
-io.smallrye:smallrye-open-api-model:4.0.1
+io.smallrye:smallrye-open-api-jaxrs:4.0.2
+io.smallrye:smallrye-open-api-model:4.0.2
 io.zipkin.reporter2:zipkin-reporter:2.16.4
 io.zipkin.reporter2:zipkin-reporter:3.3.0
 io.zipkin.reporter2:zipkin-sender-okhttp3:2.16.4

--- a/dev/io.openliberty.io.smallrye.openapi4.core/bnd.bnd
+++ b/dev/io.openliberty.io.smallrye.openapi4.core/bnd.bnd
@@ -35,7 +35,7 @@ Export-Package: \
     io.smallrye.openapi.spi.*;version=4.0,\
 
 -buildpath: \
-    io.smallrye:smallrye-open-api-core;version=4.0.1,\
+    io.smallrye:smallrye-open-api-core;version=4.0.2,\
     io.openliberty.io.smallrye.openapi4.model;version=latest,\
     io.openliberty.jakarta.validation.3.0;version=latest,\
     io.openliberty.org.eclipse.microprofile.openapi.4.0;version=latest,\

--- a/dev/io.openliberty.io.smallrye.openapi4.jaxrs/bnd.bnd
+++ b/dev/io.openliberty.io.smallrye.openapi4.jaxrs/bnd.bnd
@@ -23,7 +23,7 @@ javac.target: 11
 Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=11))"
 
 Include-Resource: \
-    @${repo;io.smallrye:smallrye-open-api-jaxrs;4.0.1}!/META-INF/services/*
+    @${repo;io.smallrye:smallrye-open-api-jaxrs;4.0.2}!/META-INF/services/*
 
 Service-Component: \
     io.smallrye.openapi.runtime.scanner.spi.AnnotationScanner; \
@@ -40,7 +40,7 @@ Export-Package: \
     io.smallrye.openapi.jaxrs;version=4.0;thread-context=true
 
 -buildpath: \
-    io.smallrye:smallrye-open-api-jaxrs;version=4.0.1,\
+    io.smallrye:smallrye-open-api-jaxrs;version=4.0.2,\
     io.openliberty.io.smallrye.openapi4.core;version=latest,\
     io.openliberty.jakarta.validation.3.0;version=latest,\
     io.openliberty.org.eclipse.microprofile.openapi.4.0;version=latest,\

--- a/dev/io.openliberty.io.smallrye.openapi4.model/bnd.bnd
+++ b/dev/io.openliberty.io.smallrye.openapi4.model/bnd.bnd
@@ -27,7 +27,7 @@ Export-Package: \
     io.smallrye.openapi.model.*;version=4.0
 
 -buildpath: \
-    io.smallrye:smallrye-open-api-model;version=4.0.1,\
+    io.smallrye:smallrye-open-api-model;version=4.0.2,\
     io.openliberty.org.eclipse.microprofile.openapi.4.0;version=latest,\
     io.openliberty.io.smallrye.jandex3;version=latest,\
     com.ibm.ws.org.jboss.logging;version=latest

--- a/dev/io.openliberty.microprofile.openapi.2.0.internal_fat/fat/src/io/openliberty/microprofile/openapi20/fat/deployments/DeploymentTest.java
+++ b/dev/io.openliberty.microprofile.openapi.2.0.internal_fat/fat/src/io/openliberty/microprofile/openapi20/fat/deployments/DeploymentTest.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2020, 2023 IBM Corporation and others.
+ * Copyright (c) 2020, 2024 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License 2.0
  * which accompanies this distribution, and is available at
@@ -212,6 +212,16 @@ public class DeploymentTest {
         assertRest();
         assertOpenApiDoc();
         assertCache(war, CacheUsed.CACHE_NOT_USED, CacheWritten.CACHE_NOT_WRITTEN);
+    }
+
+    @Test
+    public void testJsonStaticFileWithTabs() throws Exception {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "testWar.war")
+                                   .addAsManifestResource(DeploymentTest.class.getPackage(), "static-file-with-tabs.json", "openapi.json");
+
+        deployApp(war);
+
+        assertOpenApiDoc();
     }
 
     @Test

--- a/dev/io.openliberty.microprofile.openapi.2.0.internal_fat/fat/src/io/openliberty/microprofile/openapi20/fat/deployments/static-file-with-tabs.json
+++ b/dev/io.openliberty.microprofile.openapi.2.0.internal_fat/fat/src/io/openliberty/microprofile/openapi20/fat/deployments/static-file-with-tabs.json
@@ -1,0 +1,48 @@
+{
+	"openapi": "3.0.0",
+	"info": {
+		"title": "Test REST API",
+		"version": "1.0.0",
+		"description": "Test REST API in Liberty with tabs in json",
+		"termsOfService": "http://example.com/terms/",
+		"contact":{
+		   "name": "Open API support team",
+		   "email": "oapisupport@example.com",
+		   "url": "http://www.example.com/support"
+		}
+	},
+	"paths": {
+        "/test": {
+            "get": {
+                "summary": "openapi.json - Get test data",
+                "responses": {
+                    "200": {
+                        "description": "Test data retrieved",
+                        "schema": {
+                            "$ref": "#/components/schemas/TestData"
+                        }
+                    },
+                    "404": {
+                        "description": "Test data not found"
+                    }
+                }
+            }
+				}
+		},
+    "components": {
+        "schemas": {
+            "TestData": {
+                "properties": {
+                    "fieldA": {
+                        "type": "string",
+                        "example": "foo"
+                    },
+                    "fieldB": {
+                        "type": "string",
+                        "example": "bar"
+                    }
+                }
+            }
+        }
+    }
+}

